compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -237,6 +237,16 @@ fn run_compiler(
         Box<dyn FnOnce(&config::Options) -> Box<dyn CodegenBackend> + Send>,
     >,
 ) -> interface::Result<()> {
+    // Throw away the first argument, the name of the binary.
+    // In case of at_args being empty, as might be the case by
+    // passing empty argument array to execve under some platforms,
+    // just use the original slice.
+    //
+    // This situation was possible before due to arg_expand_all being
+    // called before removing the argument, enabling a crash by calling
+    // the compiler with @empty_file as argv[0] and no more arguments.
+    let at_args = at_args.get(1..).unwrap_or(at_args);
```

Comment:
I edited the suggestion, but you applied it before reloading I guess, `get(1..)` returns `None` only if `at_args` is empty, so this can be simplified to `.unwrap_or_default()` (the comment should be similarly fixed — `just use an empty slice` or something like that).